__tests__/script.test.js

```diff
@@ -8,13 +8,20 @@ describe('Test suite for the SDK', () => {
   const xhrMock = {
```

Comment:
Add unit tests (not integration tests) for all the newly added code.

Reply:
Existing test cases are covering most of the things.

Reply:
Not enough. A lot of scenarios are not taken care of in them. Those are e2e tests.

Reply:
Addressed